Update Wave sidebar order and naming (#868)
seqeralabs/wave: docs/get-started/nextflow-wave.mdx

---
title: Nextflow and Wave
description: Provision containers with Wave directly from your Nextflow pipelines
---

You can provision containers with Wave directly from your Nextflow pipelines.

**Prerequisites**

- You have an account with a container registry, such as DockerHub, and an access token that provides write access to your container repository.

> - You have a [Seqera Cloud](https://cloud.seqera.io/login) or Platform account, to store your container registry credentials for Wave to use.

Author: Good point - updated in PR

- You have either [Docker Desktop] or [Podman] installed locally.
- You have [Nextflow] 23.10.x or newer installed locally.

In this guide, you'll build a container from a Nextflow module and freeze that package in your private container repository.

## Create your Seqera access token

1. Log in to Seqera.
1. From your personal workspace: Go to the user menu and select **Settings > Your tokens**.
1. Select **Add token**.
1. Enter a unique name for your token, then select **Add**.
1. Copy and store your token securely.

    :::caution

>     The access token is displayed only once. Save the token value before closing the **Personal Access Token** window.

Reviewer: ```suggestion
    The access token is displayed only once. Save the token value before you close the **Personal Access Token** window.
```

    :::

1. In a terminal window, assign your access token to the `TOWER_ACCESS_TOKEN` environment variable:

    ```
    export TOWER_ACCESS_TOKEN=<token_token>
    ```

## Add your container registry credentials to Seqera

When freezing a container to the build repository that you specify, Wave uses Seqera to obtain your registry access credentials. If you use Docker as your container registry, complete the following steps:

To create your access token in Docker Hub:

1. Log in to [Docker Hub](https://hub.docker.com/).
2. Select your username in the top right corner and select **Account Settings**.
3. Select **Security > New Access Token**.

> 4. Enter a token description and select **Read-only** from the Access permissions drop-down menu, then select **Generate**.

Reviewer: ```suggestion
4. Enter a token description. Select **Access permissions > Read-only**, then select **Generate**.
```

5. Copy and save the generated access token (this is only displayed once).

To add your credentials to Seqera:

1.  Add your credentials to your organization or personal workspace:
    - From an organization workspace: Go to **Credentials > Add Credentials**.
    - From your personal workspace: From the user menu, go to **Your credentials > Add credentials**.

2.  Complete the following fields:

    - **Name**: Specify a unique name for the credentials using alphanumeric characters, dashes, or underscores. For example, `my-registry-creds`.
    - **Provider**: Select **Container registry**.
    - **User name**: Specify your Docker username. For example, `user1`.
    - **Password**: Specify your personal access token (PAT). For example, `1fcd02dc-...215bc3f3`.
    - **Registry server**: Specify the container registry hostname, excluding the protocol. For example, `docker.io`.

> 3.  After you've completed all the form fields, select **Add**. The new credential is now listed under the **Credentials** tab.

Reviewer: ```suggestion
3.  Select **Add**. The new credential will be listed under the **Credentials** tab.
```
